chainer/functions/loss/softmax_cross_entropy.py

```diff
@@ -48,6 +54,12 @@ def forward_cpu(self, inputs):
         log_y = log_softmax._log_softmax(x, self.use_cudnn)
         if self.cache_score:
             self.y = numpy.exp(log_y)
+        if self.class_weight is not None:
+            if self.class_weight.shape != x.shape:
+                shape = [1 if d != 1 else -1 for d in six.moves.range(x.ndim)]
+                self.class_weight = numpy.broadcast_to(
+                    self.class_weight.reshape(shape), x.shape)
+            log_y *= self.class_weight
```

Comment:
It seems that

1. l.58 is no needed.
2. It is undesirable to implicitly change the shape of the public variable `class_weight`. 
How about using the following?

```python
if self.class_weight is not None:
    shape = [1 if d != 1 else -1 for d in six.moves.range(x.ndim)]
    log_y *= numpy.broadcast_to(
        self.class_weight.reshape(shape), x.shape)
```